.github/workflows/nightly.yml

```diff
@@ -89,3 +89,31 @@ jobs:
           cargo hakari disable
           cargo hakari remove-deps -y
           RUSTC_BOOTSTRAP=1 cargo udeps
+
+  report-status:
+    name: Report Status
+    runs-on: [self-hosted, self-hosted-ubuntu]
+    if: always()
+    needs: [beta, release, cargo-udeps]
+    steps:
+      - name: Post to slack
+        uses: slackapi/slack-github-action@v1.19.0
+        uses: technote-space/workflow-conclusion-action@v3
+        env:
+          SLACK_BOT_TOKEN: ${{ secrets.SLACK_BOT_TOKEN }}
+        with:
+          channel-id: 'build-failures-sui'
+          payload: |
+            {
+              "text": "Workflow ${{ github.workflow }} failed\nLogs are here: ${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }}"
+              "blocks": [
+                {
+                  "type": "section",
+                  "text": {
+                    "type": "mrkdwn",
+                    "text": "Workflow ${{ github.workflow }} failed\nLogs are here: ${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }}"
+                  }
+                }
+              ]
+            }
+        if: env.WORKFLOW_CONCLUSION == 'failure' # notify only if failure
```

Comment:
Maybe its clearer if this if/conditional is further up, for example before or after env?

Reply:
Sounds good, I moved it.